File: docs/guides/php/src/setHeaderUserIDThenSaveObjects.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;

$playlists = [/* Your records */];

$client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

foreach ($playlists as $playlist) {
    try {
        $playlistUserID = $playlist['userID'];
        $client->saveObjects(
            '<YOUR_INDEX_NAME>',
            $playlists,
            false,
            1000,
            [
                'headers' => [
                    'X-Algolia-User-ID' => $playlistUserID,
                ],
            ]
        );
    } catch (Exception $e) {
        echo $e->getMessage().PHP_EOL;
    }
}

File: docs/guides/ruby/setHeaderUserIDThenSaveObjects.rb
require "algolia"

# Your records
playlists = []

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

playlists.each do |playlist|
  playlist_user_id = playlist["userID"]
  client.save_objects(
    "<YOUR_INDEX_NAME>",
    playlists,
    false,
    1000,
    {:header_params => {"X-Algolia-User-ID" => playlist_user_id}}
  )
end
